### ISSUE TITLE
no such package '@nodejs//

### ISSUE THREAD

**Issue by pward123** [bazelbuild/rules_nodejs#346]
I'm having a problem getting started. I keep getting "no such package '@nodejs//" errors.

I'm sure it's something I'm doing wrong, so I created a reproduction in the following repo:
https://github.com/pward123/bazel-native-test

Any help would be greatly appreciated. Thanks!

**Comment by alexeagle** [comment]
Bazel tends to report errors that are chained from other errors. In your case the proximate error is `Unable to load package for //:package.json: not found` which is because there is no BUILD/BUILD.bazel file in the same directory with package.json, meaning there is no Bazel package there.

**Comment by pward123** [comment]
Thanks for your assistance. 

I've tried moving the package.json to the packages/my-app folder and I've tried adding a package.json while leaving the repo root file in place. Both end up giving me essentially the same error.

**Comment by pward123** [comment]
Looks like there were a number of things that were wrong. I've pushed the resolution to the repo.

For future surfers, here's a list of observations:

* nodejs_binary entry_point needs to be an absolute path rooted from the workspace name (i.e. "workspace_name/packages/my-app/index.js"). errors messages for this are pretty good and I remembered reading this in the docs once I realized what I'd done
* workspace name cannot contain dashes. the error you get for violating this is very unintuitive
* a BUILD file is required in the same folder as the WORKSPACE file. it can be empty, but it must exist. the error for violating this very unintuitive
